### PR TITLE
[ESLint] Add `app` dir to default linting directories

### LINKED ISSUE #44424: `next lint` omits `app` dir by default.
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 22.2.0: Fri Nov 11 02:03:51 PST 2022; root:xnu-8792.61.2~4/RELEASE_ARM64_T6000
Binaries:
  Node: 18.4.0
  npm: 8.12.1
  Yarn: 1.22.19
  pnpm: 7.20.0
Relevant packages:
  next: 13.1.1
  eslint-config-next: 13.1.1
  react: 18.2.0
  react-dom: 18.2.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true), ESLint (eslint-config-next)

### Link to the code that reproduces this issue

-

### To Reproduce

Just run the `next lint` command when you have some issues inside `app` directory. They will be skipped by default as `app` dir is not listed inside `ESLINT_DEFAULT_DIRS` 

### Describe the Bug

By default, the `next lint` omits the `app` directory.

### Expected Behavior

Linter should also go through the `app` directory by default.

### Which browser are you using? (if relevant)

_No response_

### How are you deploying your application? (if relevant)

_No response_

### PR DESCRIPTION
**Pull request by wojtekolek**
Fixes #44424 by adding the `app` folder to an `ESLINT_DEFAULT_DIRS` constant which defines all folders where the linter should go through.

## Bug

- [x] Related issues linked using `fixes #number`
- [x] Integration tests added
- [ ] Errors have a helpful link attached, see [`contributing.md`](https://github.com/vercel/next.js/blob/canary/contributing.md)

### CODE REVIEW

File: packages/next/lib/constants.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ export const NON_STANDARD_NODE_ENV = `You are using a non-standard "NODE_ENV" va
 
 export const SSG_FALLBACK_EXPORT_ERROR = `Pages with \`fallback\` enabled in \`getStaticPaths\` can not be exported. See more info here: https://nextjs.org/docs/messages/ssg-fallback-true-export`
 
-export const ESLINT_DEFAULT_DIRS = ['pages', 'components', 'lib', 'src']
+export const ESLINT_DEFAULT_DIRS = ['pages', 'app', 'components', 'lib', 'src']
```
Comment (ijjk) [inline packages/next/lib/constants.ts:42]: Can we only add the `app` dir here when `experimental.appDir` is enabled and add a note to make default when stable?

**Review by ijjk — COMMENTED** [review]
@janicklas-ralph @housseindjirdeh Do you have any thoughts on this change? Are there any lint rules we need to consider that may conflict with `app`?

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | wojtekolek/next.js issue/#44424 | Change |
| - | - | - | - |
| buildDuration | 36.9s | 37.2s | ⚠️ +276ms |
| buildDurationCached | 10.4s | 11.1s | ⚠️ +727ms |
| nodeModulesSize | 101 MB | 101 MB | ✓ |
| nextStartRea..uration (ms) | 283ms | 276ms | -7ms |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | wojtekolek/next.js issue/#44424 | Change |
| - | - | - | - |
| 152-HASH.js gzip | 64.7 kB | 64.7 kB | ✓ |
| 53.HASH.js gzip | 182 B | 182 B | ✓ |
| main-app-HASH.js gzip | 205 B | 205 B | ✓ |
| main-HASH.js gzip | 79.2 kB | 79.2 kB | ✓ |
| webpack-HASH.js gzip | 1.7 kB | 1.7 kB | ✓ |
| Overall change | 146 kB | 146 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | wojtekolek/next.js issue/#44424 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | wojtekolek/next.js issue/#44424 | Change |
| - | - | - | - |
| _app-HASH.js gzip | 192 B | 192 B | ✓ |
| _error-HASH.js gzip | 179 B | 179 B | ✓ |
| amp-HASH.js gzip | 484 B | 484 B | ✓ |
| css-HASH.js gzip | 804 B | 804 B | ✓ |
| dynamic-HASH.js gzip | 2.29 kB | 2.29 kB | ✓ |
| edge-ssr-HASH.js gzip | 255 B | 255 B | ✓ |
| head-HASH.js gzip | 827 B | 827 B | ✓ |
| hooks-HASH.js gzip | 848 B | 848 B | ✓ |
| image-HASH.js gzip | 4.3 kB | 4.3 kB | ✓ |
| index-HASH.js gzip | 252 B | 252 B | ✓ |
| link-HASH.js gzip | 2.69 kB | 2.69 kB | ✓ |
| routerDirect..HASH.js gzip | 782 B | 782 B | ✓ |
| script-HASH.js gzip | 857 B | 857 B | ✓ |
| withRouter-HASH.js gzip | 781 B | 781 B | ✓ |
| 85e02e95b279..7e3.css gzip | 107 B | 107 B | ✓ |
| Overall change | 15.6 kB | 15.6 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | wojtekolek/next.js issue/#44424 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 480 B | 480 B | ✓ |
| Overall change | 480 B | 480 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | wojtekolek/next.js issue/#44424 | Change |
| - | - | - | - |
| index.html gzip | 488 B | 488 B | ✓ |
| link.html gzip | 503 B | 503 B | ✓ |
| withRouter.html gzip | 484 B | 484 B | ✓ |
| Overall change | 1.48 kB | 1.48 kB | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | wojtekolek/next.js issue/#44424 | Change |
| - | - | - | - |
| edge-ssr.js gzip | 110 kB | 110 kB | ✓ |
| page.js gzip | 98.5 kB | 98.5 kB | -2 B |
| Overall change | 209 kB | 209 kB | -2 B |

</details>

<details>
<summary><strong>Middleware size</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | wojtekolek/next.js issue/#44424 | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 585 B | 583 B | -2 B |
| middleware-r..fest.js gzip | 145 B | 145 B | ✓ |
| middleware.js gzip | 27.1 kB | 27.1 kB | ✓ |
| edge-runtime..pack.js gzip | 1.83 kB | 1.83 kB | ✓ |
| Overall change | 29.7 kB | 29.7 kB | -2 B |

</details>

#### Diffs
<details>
<summary>Diff for <strong>page.js</strong></summary>

```diff
@@ -8630,13 +8630,13 @@ Learn More: https://nextjs.org/docs/messages/node-module-in-edge-runtime`;
       /***/
     },
 
-    /***/ 5722: /***/ (
+    /***/ 9074: /***/ (
       __unused_webpack_module,
       __unused_webpack_exports,
       __webpack_require__
     ) => {
       Promise.resolve(/* import() eager */).then(
-        __webpack_require__.bind(__webpack_require__, 1210)
+        __webpack_require__.bind(__webpack_require__, 6448)
       );
       Promise.resolve(/* import() eager */).then(
         __webpack_require__.bind(__webpack_require__, 4916)
@@ -8645,7 +8645,7 @@ Learn More: https://nextjs.org/docs/messages/node-module-in-edge-runtime`;
         __webpack_require__.bind(__webpack_require__, 9916)
       );
       Promise.resolve(/* import() eager */).then(
-        __webpack_require__.bind(__webpack_require__, 6448)
+        __webpack_require__.bind(__webpack_require__, 1210)
       );
 
       /***/
```
</details>
<details>
<summary>Diff for <strong>middleware-b..-manifest.js</strong></summary>

```diff
@@ -9,7 +9,7 @@ self.__BUILD_MANIFEST = {
   rootMainFiles: [
     "static/chunks/webpack-c452a3e31b73f504.js",
     "static/chunks/152-a496ee85b6017517.js",
-    "static/chunks/main-app-08cd9540282ed02d.js"
+    "static/chunks/main-app-5259dd53eb6852cf.js"
   ],
   pages: {
     "/": [
```
</details>
<details>
<summary>Diff for <strong>main-app-HASH.js</strong></summary>

```diff
@@ -1,13 +1,13 @@
 (self["webpackChunk_N_E"] = self["webpackChunk_N_E"] || []).push([
   [744],
   {
-    /***/ 207: /***/ function(
+    /***/ 8795: /***/ function(
       __unused_webpack_module,
       __unused_webpack_exports,
       __webpack_require__
     ) {
       Promise.resolve(/* import() eager */).then(
-        __webpack_require__.t.bind(__webpack_require__, 429, 23)
+        __webpack_require__.t.bind(__webpack_require__, 8138, 23)
       );
       Promise.resolve(/* import() eager */).then(
         __webpack_require__.t.bind(__webpack_require__, 1161, 23)
@@ -16,7 +16,7 @@
         __webpack_require__.t.bind(__webpack_require__, 9379, 23)
       );
       Promise.resolve(/* import() eager */).then(
-        __webpack_require__.t.bind(__webpack_require__, 8138, 23)
+        __webpack_require__.t.bind(__webpack_require__, 429, 23)
       );
 
       /***/
@@ -28,7 +28,7 @@
       return __webpack_require__((__webpack_require__.s = moduleId));
     };
     /******/ __webpack_require__.O(0, [152], function() {
-      return __webpack_exec__(7070), __webpack_exec__(207);
+      return __webpack_exec__(7070), __webpack_exec__(8795);
     });
     /******/ var __webpack_exports__ = __webpack_require__.O();
     /******/ _N_E = __webpack_exports__;
```
</details>
</details>
Commit: e3796088b4da7e190eda6b814289ca52ae00accb

File: packages/next/lib/constants.ts (hunk 1 above)
Comment (wojtekolek) [reply packages/next/lib/constants.ts]: Sure @ijjk ! I added a new const and some if's for a proper selection 😄 Thanks for pointing this out! 🙇

File: docs/basic-features/eslint.md [hunk 2]
```diff
@@ -126,7 +126,7 @@ If you're using `eslint-plugin-next` in a project where Next.js isn't installed
 
 ## Linting Custom Directories and Files
 
-By default, Next.js will run ESLint for all files in the `pages/`, `components/`, `lib/`, and `src/` directories. However, you can specify which directories using the `dirs` option in the `eslint` config in `next.config.js` for production builds:
+By default, Next.js will run ESLint for all files in the `pages/`, `app`(if enabled), `components/`, `lib/`, and `src/` directories. However, you can specify which directories using the `dirs` option in the `eslint` config in `next.config.js` for production builds:
```
Comment (SukkaW) [inline docs/basic-features/eslint.md:129]:
```suggestion
By default, Next.js will run ESLint for all files in the `pages/`, `app` (only if the experimental `appDir` feature is enabled), `components/`, `lib/`, and `src/` directories. However, you can specify which directories using the `dirs` option in the `eslint` config in `next.config.js` for production builds:
```

File: docs/basic-features/eslint.md [hunk 3]
```diff
@@ -126,7 +126,7 @@ If you're using `eslint-plugin-next` in a project where Next.js isn't installed
 
 ## Linting Custom Directories and Files
 
-By default, Next.js will run ESLint for all files in the `pages/`, `components/`, `lib/`, and `src/` directories. However, you can specify which directories using the `dirs` option in the `eslint` config in `next.config.js` for production builds:
+By default, Next.js will run ESLint for all files in the `pages/`, `app` (only if the experimental `appDir` feature is enabled), `components/`, `lib/`, and `src/` directories. However, you can specify which directories using the `dirs` option in the `eslint` config in `next.config.js` for production builds:
```
Comment (mizdra) [inline docs/basic-features/eslint.md:129]: A list of lint target directories can also be found below. I think this one should be modified as well.

https://github.com/vercel/next.js/blob/d4a3c66479f6ea48c958c663067800d5e3fecf21/docs/api-reference/cli.md?plain=1#L123

Comment (wojtekolek) [reply docs/basic-features/eslint.md:129]: Good catch @mizdra ! Thanks and done! 😄

**Review by feedthejim — APPROVED** [review]
looks reasonable, thanks for your work!

**Review by ijjk — APPROVED** [review]
Thanks!

**Pull request merged**